scipy/signal/_spectral_py.py

```diff
@@ -282,6 +290,10 @@ def lombscargle(
         # determinate of the system of linear equations
         D = CC * SS - CS * CS
 
+        # eps is used to prevent spurious numerical issues around the "pseudo-Nyquist"
+        if D == 0:
```

Comment:
This almost never returns true. That's why your test needs to have `np.abs(D) < eps` or something like that to account for FP cancellations.

Reply:
I should add: I've never had a DivideByZero error in the 15 years I've been using lombscargle (across several variations I've written in several languages). That's how rare this issue is. And I'm always using frequencies that span across (from below to above) the "psuedo-Nyquist", which is where this could occur.

Reply:
Oops. I need to multiply eps by the sign of D (since signed zero is a thing).

Reply:
The following code produces valid results, even if numerator and denominator are  both small (e.g., multiplying numerator and denominator by 1e-20 does produce a loss in accuracy for floating point values):
```python

a_numerator, b_numerator = (YC * SS - YS * CS), (YS * CC - YC * CS)
a[i] = a_numerator / D if abs(D) > abs(a_numerator)*eps else a_numerator / eps
b[i] = b_numerator / D if abs(D) > abs(b_numerator)*eps else b_numerator / eps
```

Reply:
I'll have to play around with this, since it isn't immediately clear to me why you're comparing D with numerator*eps. Regardless, it would still need `* math.copysign(D)` added to the else condition.

I still think trying to only eliminate the division-by-zero possibility is most-targeted solution, since it is the only condition that would crash the code (and it's just a numerical edge case consequence, not an incorrect choice of inputs by the user). Changing the formulas otherwise to prevent or change very large/small values, is just adding unnecessary preference to what we think appropriate values should be.